src/cbapi/connection.py

```diff
@@ -94,8 +96,10 @@ def __init__(self, credentials, integration_name=None):
         if not credentials.ssl_verify_hostname:
             self.session.mount(self.server, HostNameIgnoringAdapter())
 
+        self._timeout = timeout
+
         # TODO: apply this to the ssl_verify_hostname case as well
-        self.session.mount(self.server, HTTPAdapter(max_retries=MAX_RETRIES))
+        self.session.mount(self.server, HTTPAdapter(max_retries=ZERO_RETRIES if self._timeout else MAX_RETRIES))
```

Comment:
@jgarman I tested the API with `timeout=10, max_retries=Retry(0, False)` and it timed out as expected.  Thanks!